src/Sfa.Tl.ResultsAndCertification.Web/Controllers/AdminDashboardController.cs

```diff
@@ -451,8 +451,66 @@ public async Task<IActionResult> AdminReviewChangesIndustryPlacementAsync(AdminR
             }
             else { return RedirectToAction(RouteConstants.ProblemWithService); }
 
-           
+
         }
         #endregion
+
+        #region Assesment Entry
+
+        [HttpGet]
+        [Route("admin/add-assessment-entry-core/{registrationPathwayId}", Name = RouteConstants.AdminCoreComponentAssessmentEntry)]
+        public async Task<IActionResult> AdminCoreComponentAssessmentEntry(int registrationPathwayId)
+        {
+            var viewModel = await _loader.GetAdminLearnerRecordWithCoreComponents(registrationPathwayId);
+
+            if (viewModel == null)
+                return RedirectToRoute(RouteConstants.PageNotFound);
+
+            return View(viewModel);
+        }
+
+        [HttpPost]
+        [Route("admin/submit-add-assessment-entry-core", Name = RouteConstants.SubmitCoreComponentAssessmentEntry)]
+        public async Task<IActionResult> AdminCoreComponentAssessmentEntry(AdminCoreComponentViewModel model)
+        {
+            var adminCoreComponent = await _loader.GetAdminLearnerRecordWithCoreComponents(model.RegistrationPathwayId);
```

Comment:
Why do you have call the loader when the model comes as a parameter already?

Reply:
There is data on the model which is missing on the post - which can be added to the form as hidden items or in the cache. However that can impact the page performance. This  seem to be a better way to implement.